# **140 python solved questions**

### 1.Write a program to print "Hello Python"

In [ ]:
print("Hello Python")

Hello Python


Hello Python


### 2.Write a program to do arithmatic operations

In [ ]:
a=float(input("Enter the first number: "))
b= float(input("Enter the second number: "))
sum = a+b
sub = a-b
mul = a*b
div = a/b
print("Addition of a and b is : ",sum)
print("subtraction of a and b is : ",sub)
print("Multiplication of a and b is : ",mul)
print("Divison of a and b is : ",div)

#input given a=10 , b=10


Addition of a and b is :  20.0
subtraction of a and b is :  0.0
Multiplication of a and b is :  100.0
Divison of a and b is :  1.0


### 3.Write a python program to find area of triangle

In [7]:
b=float(input("Enter the base: "))
h=float(input("Enter the height: "))
area = 0.5*b*h
print("Area of triangle is : ",area)

#input given b=10,h=15



Area of triangle is :  75.0


### 4.Write a python program to swap two variables

In [13]:
a =10
b = 20
temp = a
a = b 
b = temp
print("a:",a,"b:",b)

a: 20 b: 10


### 5.Write a python program to generate a random number

In [15]:
import random
print(f"random number: {random.randint(1,100)}")

random number: 77


### 6.Write a python program to convert kilometers to miles

In [23]:
kilometers = 100
# 1 kilometer = 0.621371 miles
miles = kilometers*0.621371
print(f"{kilometers} kilometer is {miles} miles")

100 kilometer is 62.137100000000004 miles


### 7.Write a python program to convert Celcius to Fahrenheit

In [ ]:
celcius = float(input("Enter the temperature in celcius: "))
# Fahrenheit = (celcius*9/5)+32
Fahrenheit = (celcius*9/5)+32
print(f"{celcius} degrees celcius is {Fahrenheit}F") 

#input given: 37

37.0degrees celcius is 98.6F


### 8.Write a program to display calender

In [29]:
import calendar
year = 2023
month = 11
cal = calendar.month(year,month)
print(cal)

   November 2023
Mo Tu We Th Fr Sa Su
       1  2  3  4  5
 6  7  8  9 10 11 12
13 14 15 16 17 18 19
20 21 22 23 24 25 26
27 28 29 30



### 9.Write a python program to solve quadratic equations

In [32]:
# standard form = ax^2+bx+c=0 where a,b,c are real numbers and a != 0 
import math
a = float(input("Enter coeffecient of a: "))
b = float(input("Enter coeffecient of b: "))
c = float(input("Enter coeffecient of c: "))
 
discriminant = b**2-4*a*c 
if discriminant > 0:
    root1 = (-b + math.sqrt(discriminant)) / (2*a)
    root2 = (-b - math.sqrt(discriminant)) / (2*a)
    print("Root1 = ",root1)
    print("Root2 = ",root2)
elif discriminant == 0:
    root = -b / (2*a)
    print("root = ",root)
else : 
    real_part = -b / (2*a)   
    img_part = math.sqrt(abs(discriminant)) / (2*a)  
    print(f"Root1 = {real_part} + {img_part}i")
    print(f"Root1 = {real_part} - {img_part}i")
    
    
    #input given : a=1,b=4,c=8

Root1 = -2.0 + 2.0i
Root1 = -2.0 - 2.0i


### 10.Write a python program to swap two numbers without temp

In [34]:
a=10
b=20
a,b = b,a
print(f"a: {a} b: {b}")

a: 20 b: 10


### 11.Write a Python program to check if the number is positive,negative or zero

In [ ]:
num = int(input("Enter a number to check if its positive , negative or zero"))
if(num>0):
    print("its a positive number")
elif(num<0):
    print("Its a negative number")
else:
    print("its a zero")
    
    ##input: 10
        

its a positive number


### 12.Write a program to check if a number is odd or even

In [37]:
num = int(input("Enter a number"))
if(num%2==0):
    print("its a even number")
else:
    print("its an odd number")
 
 #input:18       
    

its a even number


### 13.Write a python program to check Leap year

In [39]:
year = int(input("enter a year: "))
#divided by 100 means century year(ending with 00)
#century year divided by 400 is a leap year
if(year % 400 == 0) and (year % 100 == 0):
    print("{0} is a leap year".format(year))
    
# not divided by 100 meeans not a century year
# year divided by 4 is a leap year 

#century year divided by 400 is a leap year
if(year % 4 == 0) and (year % 100 != 0):
    print("{0} is a leap year".format(year))
else:
  print("{0} is not a leap year".format(year))   
  
#input: 2024  

 

2024 is a leap year


### 14.Write a python program to check prime number

In [ ]:
num = int(input("Enter a number: "))
if num==1:
    print(f"{num} is not a prime number")
elif(num > 1):
    for i in range(2,num):
        if(num % i) == 0:
            print("Its not a prime number") 
            break
    else:
        print("its a prime number")
else:
    print("please enter a positive number")
    
    #input: 23                   

its a prime number


### 15.Write a Program to find prime numbers in an interval of 1-10

In [47]:
lower = 1
upper = 10

print("prime numbers between",lower,"and",upper,"are ")
for num in range(lower,upper+1):
    if(num>1):
        for i in range(2,num):
            if(num%i)==0:
                break
        else:
            print(num)
            


prime numbers between 1 and 10 are 
2
3
5
7


### 16.Write a program to find the factorial of a number

In [ ]:
a = int(input("Enter a number: "))
factorial = 1
if( a == 0):
    print(factorial)
elif a<0:
    print("factorial does not exist for negative numbers")
else:
    for i in range(1,a+1):
        factorial = factorial*i
    print("factorial of",a,"is",factorial)        

#input:5

factorial of 5 is 120


### 17.Write a program to display the multiplication table

In [ ]:
num = int(input("Enter a number"))
mul = 1
for i in range(1,11):
    mul = num*i
    print(num,"X",i,"=",mul)
    
 #input: 2   
    

2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18
2 X 10 = 20


### 18.Write a program to print fibonacci series

In [ ]:
num = int(input("How many terms? "))
n1,n2 = 0,1
count = 0
if num<=0:
    print("enter a postive number ")
elif num == 1:
    print("Fibonacii sequence upto",num,":")
    print(n1)
else:
    print("Fibonacci series:")
    while count < num:
        print(n1)
        nth = n1 + n2
        #update values
        n1 = n2
        n2 = nth 
        count += 1       
    
#input:10    


Fibonacci series:
0
1
1
2
3
5
8
13
21
34


### 19.Write a program to chcek Armstrong Number.

In [ ]:
num = int(input("Enter a number: "))

#calculate the number of digoits in num
num_str = str(num)
num_digits = len(num_str)

#initialize variables
sum_of_powers = 0
temp = num

#calculate the sum of digits eaised to the power of num_digits
while temp > 0:
    digit = temp % 10
    sum_of_powers += digit ** num_digits
    temp //=10 
    
#chcek if its an Armstrong number

if sum_of_powers == num:
    print(f"{num} is an Armstrong number.")
else:
       print(f"{num} is not an Armstrong number.")        
 
 #input:9474
 

9474 is an Armstrong number.


<sub>
<b>Dry Run for num = 153</b><br><br>

<b>Initial values:</b><br>
num = 153<br>
num_str = "153"<br>
num_digits = 3<br>
sum_of_powers = 0<br>
temp = 153<br><br>

<b>Iteration 1:</b><br>
digit = 153 % 10 = 3<br>
sum_of_powers = 0 + 3³ = 27<br>
temp = 153 // 10 = 15<br><br>

<b>Iteration 2:</b><br>
digit = 15 % 10 = 5<br>
sum_of_powers = 27 + 5³ = 27 + 125 = 152<br>
temp = 15 // 10 = 1<br><br>

<b>Iteration 3:</b><br>
digit = 1 % 10 = 1<br>
sum_of_powers = 152 + 1³ = 152 + 1 = 153<br>
temp = 1 // 10 = 0 → loop ends<br><br>

<b>Final Check:</b><br>
sum_of_powers = 153 == num = 153 → <b>Armstrong </b>
</sub>


### 20.Write a program to find Armstrong number in an interval

In [ ]:
lower = int(input("Enter the lower interval of the number : "))
upper = int(input("Enter the upper interval of the number : "))

for num in range(lower, upper+1):
    order = len(str(num))
    temp_num = num
    sum = 0
    
    while temp_num > 0:
        digit = temp_num % 10
        sum += digit ** order
        temp_num //=10
        
    if num == sum:
        print(num)  
        
#input: lower= 10 , upper = 1000          

153
370
371
407


### 21.Program to find sum of natural numbers.

In [9]:
a = int(input("Enter the list: "))
sum = 0

for i in range(1,a+1):
    sum += i
print(sum)    


55


### 22.Python program to find LCM

In [ ]:
def compute_lcm(x , y):
    if x > y:
        greater = x
    else:
        greater = y
    while(True):
        if((greater % x == 0) and (greater % y == 0)):
            lcm = greater
            break
        greater +=1
    return lcm
num1 = int(input("Enter the first number: ")) 
num2 = int(input("Enter the second number: "))

print("The LCM is",compute_lcm(num1,num2)) 

#input: 54 and 24    
                


The LCM is 216


<sub>
<b>Dry Run for num1 = 54, num2 = 24</b><br><br>

<b>Step 1: Function call</b><br>
x = 54, y = 24<br>
x > y → greater = 54<br><br>

<b>Step 2: while(True) loop</b><br>
1st check: greater = 54 → (54 % 54 == 0) ✅ and (54 % 24 == 6) ❌ → not LCM → greater = 55<br>
2nd check: greater = 55 → (55 % 54 == 1) ❌ → greater = 56<br>
...<br>
Continue incrementing until common multiple found.<br><br>

<b>Important hits:</b><br>
greater = 108 → (108 % 54 == 0) ✅ and (108 % 24 == 12) ❌<br>
greater = 216 → (216 % 54 == 0) ✅ and (216 % 24 == 0) ✅ → LCM found<br><br>

<b>Step 3: Return</b><br>
lcm = 216 → return 216<br><br>

<b>Output:</b><br>
"The LCM is 216"
</sub>


### 23.Write a program to find HCF

In [11]:
def compute_hcf(x,y):
    if x > y:
        smaller = y
    else:
        smaller = x
    for i in range(1,smaller+1):
        if ((x % i == 0) and (y % i == 0)):
            hcf = i
    return hcf
num1 = int(input("Enter the first number: ")) 
num2 = int(input("Enter the second number: "))

print("The hcf is",compute_hcf(num1,num2)) 

#input: 54 amd 24
           

The hcf is 6


### 24.Python program to convert decimal to binary , octal and hexadecimal.

In [ ]:
dec_num = int(input("Enter a decimal number: "))

print("the decimal velue of",dec_num,"is")
print(bin(dec_num),"in binary")
print(oct(dec_num),"in octal")
print(hex(dec_num),"in hexa-decimal")

#input:27

the decimal velue of 27 is
0b11011 in binary
0o33 in octal
0x1b in hexa-decimal


### 25.Program to find Ascii value of a number

In [ ]:
char = str(input("Enter the character: "))
print("The ASCII value of " + char + " is",ord(char))

#input: P

The ASCII value of P is 80


### 26.Program to make a simple calculator with 4 basic mathematical operations

In [ ]:
def add(x,y):
    return x+y

def sub(x,y):
    return x-y

def mul(x,y):
    return x*y

def div(x,y):
    return x/y

print("select operation")
print("1.Addition \n 2.Subtraction \n 3.multiplication \n 4.Divison")  

while True:
    choice = input("Enter a choice (1/2/3/4): ")
    if choice in ('1','2','3','4'):
        try:
            num1 = float(input("Enter first number: "))
            num2 = float(input("Enter second number: ")) 
        except ValueError:
            print("Invalid input , please enter a correct number")   
            continue
        if choice == '1':
            print(num1,"+",num2,"=",add(num1,num2))
        elif choice == '2':
            print(num1,"-",num2,"=",sub(num1,num2))
            
        elif choice == '3':
            print(num1,"*",num2,"=",mul(num1,num2))
            
        elif choice == '4':
            print(num1,"/",num2,"=",div(num1,num2)) 
            
        next_calc = input("do u want to continue (yes/no): ")
        if next_calc == "no":
            break
    else:
            print("invalid input")    
            
                       
            
                  

select operation
1.Addition 
 2.Subtraction 
 3.multiplication 
 4.Divison
5.0 + 6.0 = 11.0
50.0 - 5.0 = 45.0
22.0 * 2.0 = 44.0
invalid input
invalid input
99.0 / 9.0 = 11.0


### 27.Display fibonacci series using recursion

In [27]:
def fib(n):
    if n <= 1:
        return(n)
    else: 
        return(fib(n-1) + fib(n-2))    
nterms = int(input("Enter the number of terms (greater then zero)"))
if nterms <=0:
    print("Please enter positive integer")
else:
    print("Fibonacci series:")
    for i in range(nterms):
        print(fib(i))    

Fibonacci series:
0
1
1
2
3
5
8
13
